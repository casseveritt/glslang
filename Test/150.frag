#version 150 core

in vec4 gl_FragCoord;
layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;  // ERROR
layout(pixel_center_integer) in vec4 gl_FragCoord;  // ERROR
layout(origin_upper_left) in vec4 foo;  // ERROR
layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;

void main()
{
    vec4 c = gl_FragCoord;
}

layout(origin_upper_left, pixel_center_integer) in vec4 gl_FragCoord;  // ERROR, declared after use

in struct S { float f; } s;

float patch = 3.1;