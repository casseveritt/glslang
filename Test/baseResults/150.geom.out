150.geom
Warning, version 150 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:15: 'fromVertex' : block instance name redefinition 
ERROR: 0:19: 'fromVertex' : redefinition 
ERROR: 0:21: 'fooC' : block instance name redefinition 
ERROR: 0:29: 'EmitStreamVertex' : no matching overloaded function found 
ERROR: 0:30: 'EndStreamPrimitive' : no matching overloaded function found 
ERROR: 0:44: 'stream' : can only be used on an output 
ERROR: 0:45: 'stream' : can only be used on an output 
ERROR: 0:46: 'stream' : can only be used on an output 
ERROR: 0:47: 'stream' : can only be used on an output 
ERROR: 0:47: 'stream' : can only be used on an output 
ERROR: 0:60: 'stream' : member cannot contradict block 
ERROR: 0:65: 'max_vertices' : can only apply to a standalone qualifier 
ERROR: 0:70: 'points' : cannot change previously set output primitive 
ERROR: 0:71: 'points' : cannot change previously set output primitive 
ERROR: 0:72: 'triangle_strip' : cannot apply to input 
ERROR: 0:73: 'triangle_strip' : cannot apply to: uniform
ERROR: 0:74: 'triangle_strip' : can only apply to a standalone qualifier 
ERROR: 0:75: 'triangle_strip' : can only apply to a standalone qualifier 
ERROR: 0:76: 'invocations' : not supported for this version or the enabled extensions 
ERROR: 0:76: 'invocations' : can only apply to a standalone qualifier 
ERROR: 0:78: 'invocations' : not supported for this version or the enabled extensions 
ERROR: 0:78: 'invocations' : can only apply to a standalone qualifier 
ERROR: 0:79: 'max_vertices' : can only apply to a standalone qualifier 
ERROR: 0:80: 'triangle_strip' : can only apply to a standalone qualifier 
ERROR: 0:83: 'lines' : cannot apply to 'out' 
ERROR: 0:85: 'triangles' : cannot change previously set input primitive 
ERROR: 0:86: 'triangles_adjacency' : cannot change previously set input primitive 
ERROR: 0:87: 'invocations' : not supported for this version or the enabled extensions 
ERROR: 0:88: 'max_vertices' : too large, must be less than gl_MaxGeometryOutputVertices 
ERROR: 0:91: 'stream' : member cannot contradict block 
ERROR: 0:91: 'stream' : can only be used on an output 
ERROR: 31 compilation errors.  No code generated.


invocations = 4
max_vertices = 300
input primitive = lines_adjancency
output primitive = triangle_strip
ERROR: node is still EOpNull!
0:25  Function Definition: main( (void)
0:25    Function Parameters: 
0:27    Sequence
0:27      EmitVertex (void)
0:28      EndPrimitive (void)
0:29      Constant:
0:29        0.000000
0:30      Constant:
0:30        0.000000
0:32      move second child to first child (3-component vector of float)
0:32        color: direct index for structure (layout(stream=0 ) out 3-component vector of float)
0:32          '__anon__0' (layout(stream=0 ) out block{layout(stream=0 ) out 3-component vector of float color})
0:32          Constant:
0:32            0 (const uint)
0:32        color: direct index for structure (in 3-component vector of float)
0:32          direct index (block{in 3-component vector of float color})
0:32            'fromV' (in 4-element array of block{in 3-component vector of float color})
0:32            Constant:
0:32              0 (const int)
0:32          Constant:
0:32            0 (const int)
0:33      move second child to first child (float)
0:33        direct index (layout(stream=0 ) float)
0:33          gl_ClipDistance: direct index for structure (layout(stream=0 ) out unsized array of float)
0:33            '__anon__1' (layout(stream=0 ) out block{layout(stream=0 ) gl_Position 4-component vector of float gl_Position, layout(stream=0 ) gl_PointSize float gl_PointSize, layout(stream=0 ) out unsized array of float gl_ClipDistance})
0:33            Constant:
0:33              2 (const uint)
0:33          Constant:
0:33            3 (const int)
0:33        direct index (float)
0:33          gl_ClipDistance: direct index for structure (in unsized array of float)
0:33            direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:33              'gl_in' (in 4-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:33              Constant:
0:33                1 (const int)
0:33            Constant:
0:33              2 (const int)
0:33          Constant:
0:33            2 (const int)
0:34      move second child to first child (4-component vector of float)
0:34        gl_Position: direct index for structure (layout(stream=0 ) gl_Position 4-component vector of float)
0:34          '__anon__1' (layout(stream=0 ) out block{layout(stream=0 ) gl_Position 4-component vector of float gl_Position, layout(stream=0 ) gl_PointSize float gl_PointSize, layout(stream=0 ) out unsized array of float gl_ClipDistance})
0:34          Constant:
0:34            0 (const uint)
0:34        gl_Position: direct index for structure (in 4-component vector of float)
0:34          direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:34            'gl_in' (in 4-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:34            Constant:
0:34              0 (const int)
0:34          Constant:
0:34            0 (const int)
0:35      move second child to first child (float)
0:35        gl_PointSize: direct index for structure (layout(stream=0 ) gl_PointSize float)
0:35          '__anon__1' (layout(stream=0 ) out block{layout(stream=0 ) gl_Position 4-component vector of float gl_Position, layout(stream=0 ) gl_PointSize float gl_PointSize, layout(stream=0 ) out unsized array of float gl_ClipDistance})
0:35          Constant:
0:35            1 (const uint)
0:35        gl_PointSize: direct index for structure (in float)
0:35          direct index (block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:35            'gl_in' (in 4-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:35            Constant:
0:35              3 (const int)
0:35          Constant:
0:35            1 (const int)
0:36      move second child to first child (int)
0:36        'gl_PrimitiveID' (layout(stream=0 ) out int)
0:36        'gl_PrimitiveIDIn' (in int)
0:37      move second child to first child (int)
0:37        'gl_Layer' (layout(stream=0 ) out int)
0:37        Constant:
0:37          2 (const int)
0:65  Function Definition: foo(i1; (void)
0:65    Function Parameters: 
0:65      'a' (in int)
0:67    Sequence
0:67      move second child to first child (4-component vector of float)
0:67        a: direct index for structure (layout(stream=6 ) out 4-component vector of float)
0:67          'ouuaa6' (layout(stream=6 ) out block{layout(stream=6 ) out 4-component vector of float a})
0:67          Constant:
0:67            0 (const int)
0:67        Constant:
0:67          1.000000
0:67          1.000000
0:67          1.000000
0:67          1.000000
0:?   Linker Objects
0:?     'fromV' (in 4-element array of block{in 3-component vector of float color})
0:?     'toF' (layout(stream=0 ) out block{layout(stream=0 ) out 3-component vector of float color})
0:?     '__anon__0' (layout(stream=0 ) out block{layout(stream=0 ) out 3-component vector of float color})
0:?     'gl_in' (in 4-element array of block{in 4-component vector of float gl_Position, in float gl_PointSize, in unsized array of float gl_ClipDistance})
0:?     'ov0' (layout(stream=0 ) out 4-component vector of float)
0:?     'ov4' (layout(stream=4 ) out 4-component vector of float)
0:?     'o1v0' (layout(stream=0 ) out 4-component vector of float)
0:?     'ua' (layout(stream=3 ) uniform int)
0:?     'ibb' (layout(stream=3 column_major shared ) uniform block{layout(stream=3 column_major shared ) uniform int ua})
0:?     'ov3' (layout(stream=3 ) out 4-component vector of float)
0:?     'ouuaa6' (layout(stream=6 ) out block{layout(stream=6 ) out 4-component vector of float a})
0:?     'ouua6' (layout(stream=6 ) out block{layout(stream=6 ) out 4-component vector of float a})
0:?     'ouua7' (layout(stream=7 ) out block{layout(stream=6 ) out 4-component vector of float a})
0:?     'ov2s3' (layout(stream=3 ) out 4-component vector of float)
0:?     'badv4' (layout(stream=3 ) out 4-component vector of float)
0:?     'bad2v4' (in 4-element array of 4-component vector of float)
0:?     '__anon__1' (layout(stream=3 ) out block{layout(stream=3 ) out int a})
0:?     'outbi' (layout(stream=3 ) out block{layout(stream=3 ) out int a, layout(stream=3 ) out int b, layout(stream=3 ) out int c})
0:?     'inbi' (in 4-element array of block{layout(stream=2 ) in int a})
0:?     'insn' (in 4-element array of block{in int a15})
0:?     '__anon__2' (layout(stream=3 ) out block{layout(stream=3 ) out float f15})
0:?     '__anon__3' (layout(column_major shared ) uniform block{layout(column_major shared ) uniform bool b15})


Linked geometry stage:


invocations = 4
max_vertices = 300
input primitive = lines_adjancency
output primitive = triangle_strip

